src/engine/strat_engine/backstore/metadata.rs

```diff
@@ -36,7 +36,54 @@ pub struct BDA {
     regions: mda::MDARegions,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq)]
+enum MetadataLocation {
+    Both,
+    First,
+    Second,
+}
+
 impl BDA {
+    fn read<F>(f: &mut F) -> Result<[u8; _BDA_STATIC_HDR_SIZE]>
+    where
+        F: Read + Seek,
+    {
+        #![allow(unused_io_amount)]
+        f.seek(SeekFrom::Start(0))?;
+        let mut buf = [0u8; _BDA_STATIC_HDR_SIZE];
+
+        // FIXME: See https://github.com/stratis-storage/stratisd/pull/476
+        f.read(&mut buf)?;
+        Ok(buf)
+    }
+
+    fn write<F>(f: &mut F, bda_buf: &[u8], which: MetadataLocation) -> Result<()>
+    where
+        F: Seek + SyncAll,
+    {
+        let zeroed = [0u8; _BDA_STATIC_HDR_SIZE];
+
+        // Write 8K header. Static_Header copies go in sectors 1 and 9.
+        if which == MetadataLocation::Both || which == MetadataLocation::First {
+            f.seek(SeekFrom::Start(0))?;
+            f.write_all(&zeroed[..SECTOR_SIZE])?; // Zero 1 unused sector
+            f.write_all(bda_buf)?;
+            f.write_all(&zeroed[..SECTOR_SIZE * 7])?; // Zero 7 unused sectors
+            f.sync_all()?;
+        }
+
+        if which == MetadataLocation::Second {
```

Comment:
Nicer if this were an ```else``` block of the first ```if```, as you are doing one or the other.

Reply:
Indeed, simplified.  Thanks!